src/extension.ts

```diff
@@ -469,20 +472,45 @@ class ExtensionManager implements vscode.Disposable {
   /**
    * Drop the current kits watcher and create a new one.
    */
-  private _resetKitsWatcher() {
+  private async _resetKitsWatcher() {
     // Throw the old one away
     this._kitsWatcher.dispose();
+
+    let kitsWatcher: string[];
+    kitsWatcher = [USER_KITS_FILEPATH];
+
     // Determine whether we need to watch the workspace kits file:
     const ws_kits_path = this._workspaceKitsPath;
-    this._kitsWatcher = ws_kits_path
-        // We have workspace kits:
-        ? new MultiWatcher(USER_KITS_FILEPATH, ws_kits_path)
-        // No workspace:
-        : new MultiWatcher(USER_KITS_FILEPATH);
+    if (ws_kits_path) {
+      kitsWatcher.push(ws_kits_path);
+    }
+
+    // Determine wether we need to watch the config kits file:
+    const kit_file_path = await this._configKitsPath;
+    if (kit_file_path) {
+      kitsWatcher.push(kit_file_path);
+    }
+
+    this._kitsWatcher = new MultiWatcher(...kitsWatcher);
```

Comment:
Thinking about this some more... The MultiWatcher requires paths that use glob patterns.  The config path could be an absolute path and would need to create a file watcher using `fs.watch()` instead of the vscode API.  It seems we would either need to restrict the config path to glob patterns, or add a new class in `watcher.ts` that wraps the `fs.FSWatcher` object (similar to `IndividualWatcher`).

Reply:
I'm not sure what you mean here. How are the other paths glob patterns? I think I am missing something :-).

Reply:
I tried running your code and I'm not seeing any of the file watchers working.  The `vscode.workspace.createFileSystemWatcher` API only works for files inside the workspace folder and asks for a glob pattern as input (`MultiWatcher` creates file watchers of this type).  The user-local kits are outside the workspace too, so file watching doesn't work on that either, but that's not your bug. 😃

Reply:
Ah I see, I thought if the others work this should also work. This explains the ```_editorWatcher```  a bit better as well. Is this something you want to fix in this PR or do as something separate?

Reply:
It seems like the path of least resistance is to just change the IndividualWatcher to use [`fs.watch`](https://nodejs.org/api/fs.html#fs_fs_watch_filename_options_listener) which would fix both problems at the same time.  Is that something you'd be willing to do?

Reply:
Willing yes, I need to check on time and knowledge (this is my first piece of typescript). I'll let you know.

Reply:
Thanks for updating the PR. I was running into some problems with it on Windows.  I meant to respond back, but was thinking I might just update the branch when I'm done.  Sorry for the delay, I've been working on 0.26.0 of the cpptools extension as well and we've had a lot of stuff to coordinate with that release.

Reply:
I can imagine :-). I need to spend a bit more time on this as I am a bit unsure why tests are failing. In my eyes nothing should have changed yet as using the new watcher is an optional argument. Not 100% sure when I can make time again though.

Reply:
@bobbrow In the latest release of vs code I noticed:

> Filewatcher update (Chokidar)
During this milestone, we updated our file watcher for macOS and Linux to version 3.x, which brings improved performance, reduced memory consumption, and a smaller package size. You can learn more from the Chokidar repo and Chokidar 3 blog post. <

Will this impact this issue? I.e. am I trying to fix something that is/will be fixed?

Reply:
VS Code file watchers are still based on glob patterns which won't let you look outside of the `${workspaceFolder}`.  It would be interesting to investigate using Chokidar instead of `fs.watch` for this, but I don't think it is required.

Reply:
The multi-root PR is fixing up some of the file watcher stuff to use Chokidar.  After that goes in, I'll help merge this with those changes.

Reply:
Hi still up for this one?

Reply:
Thanks for the reminder. We do still plan to get this in.  I'm sorry it's been so long.

Reply:
Is it wise to continue on this PR or has so much changed in the meantime that it is better to start over? Some of the functions I modified have disappeared entirely and such.

Reply:
We will analyze what is best after releasing 1.4. We will finalize on this, on way or another, thank you very much for your contribution and patience.